src/ray/raylet/scheduling_resources.cc

```diff
@@ -9,39 +9,106 @@ namespace ray {
 
 namespace raylet {
 
-bool EqualsZeroEpsilon(double quantity) {
-  if ((quantity <= EPSILON) && (quantity >= -1 * EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity() {
+  resource_name_ = "";
+  resource_quantity_ = 0;
 }
 
-bool EqualsOneEpsilon(double quantity) {
-  if ((quantity <= 1 + EPSILON) && (quantity >= 1 - EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity(int resource_quantity) {
+  resource_name_ = "";
+  resource_quantity_ =
+      static_cast<int>(std::round(resource_quantity * conversion_factor_));
+}
+
+FractionalResourceQuantity::FractionalResourceQuantity(std::string resource_name,
+                                                       double resource_quantity) {
+  RAY_CHECK(resource_quantity > 0) << "Resource " << resource_name << " capacity is "
```

Comment:
also check that it is `<1`.

Reply:
We can't check that `FractionalResourceQuantity` is `<1` because the resources of a node are stored as `ResourceSets`, which are just a mapping from resource ids to `FractionalResourceQuantity`
https://github.com/ray-project/ray/blob/1592ead002086768b8ec6ea609bd9a625a1dc08d/src/ray/raylet/main.cc#L76-L85